src/rust/engine/src/externs.rs

```diff
@@ -117,19 +120,28 @@ impl Externs {
     (self.store_list)(self.context, values_clone.as_ptr(), values_clone.len() as u64, merge)
   }
 
-  pub fn project(&self, value: &Value, field: &Field, type_id: &TypeId) -> Value {
-    (self.project)(self.context, value, field, type_id)
+  pub fn project(&self, value: &Value, field: &String, type_id: &TypeId) -> Value {
```

Comment:
Use of `&String` should not be necessary: `String` is generally an "owned" type, so if you want to loan out a reference to it, you should use a `&str` (ie, "string slice"), which is more flexible.

An advantage of this is that you should be able to call `project(value, "this_is_a_literal_string")` directly, because literals are of type `&str`. For callers that have a `String`, they'd pass `my_string.as_str()`, which is a free cast.